### PR TITLE
HOCS-2793 - Add team to deployment and secret

### PR DESCRIPTION
**Pull request by philbarton**
HOCS-279, adds a team to the config of the case creator.

Also added team to the example secret.

### CODE REVIEW

**Review by zuzak — COMMENTED** [review]
Ideally you'd use a `ConfigMap` instead of a `Secret`, because, you know, it's a configuration parameter and not a secret.

(and as the Keycloak user'll be consistent across notprod environments, I'd change the `kubectl create` script into some yaml we can apply with `kd` robustly, maybe with some templating:

```yaml
{{ if .NOT_PRODUCTION }}
   key: foo
{{ else }}
    key: bar
{{ end }}
```

**Comment by philbarton** [comment]
> Ideally you'd use a `ConfigMap` instead of a `Secret`, because, you know, it's a configuration parameter and not a secret.
> 
> (and as the Keycloak user'll be consistent across notprod environments, I'd change the `kubectl create` script into some yaml we can apply with `kd` robustly, maybe with some templating:
> 
> ```yaml
> {{ if .NOT_PRODUCTION }}
>    key: foo
> {{ else }}
>     key: bar
> {{ end }}
> ```

@zuzak I did consider a ConfigMap but thought that user and group might be considered something to hide. 
If you agree that they aren't sensitive, then I'll create a ConfigMap.

File: kd/identityConfigMap.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: hocs-complaint-identity
+data:
+  CASE_CREATOR_UKVI_COMPLAINT_USER: {{.CASE_CREATOR_UKVI_COMPLAINT_USER}}
+  CASE_CREATOR_UKVI_COMPLAINT_TEAM: {{.CASE_CREATOR_UKVI_COMPLAINT_TEAM}}
+  CASE_CREATOR_UKVI_COMPLAINT_GROUP: {{.CASE_CREATOR_UKVI_COMPLAINT_GROUP}}
```
Comment (zuzak) [inline kd/identityConfigMap.yaml:8]: This works, and it's fine -- but it'd be neater if you defined the values in yaml/Go templating, rather than passing through three env vars set in Bash:

```yaml
{{if .CLUSTER_NAME == 'acp-notprod' }} # or if .NOTPROD or something
  CASE_CREATOR_UKVI_COMPLAINT_USER: 22ca514d-c205-47a0-9e38-68daee4c1299
  CASE_CREATOR_UKVI_COMPLAINT_TEAM: 08e30ffc-2087-ff3a-b19b-343a88491347
  CASE_CREATOR_UKVI_COMPLAINT_GROUP: /COMP_CCH_zqxmzQ6iEkTRw
{{else}}
  # ...
{{end}}
```

**Review by zuzak — COMMENTED** [review]
Most (all?) of our current secrets have everything defined in terms of lowercase-dash stuff like `cs-demo-ukvi-complaint-identity` in your old secret. Here we're going with uppercase-underscore env-var stuff directly, and using `envFrom` instead of `secretKeyRef`.

It's fine -- but wondering if we should keep using only one of these two styles.

File: kd/deployment.yaml [hunk 2]
```diff
@@ -102,12 +102,14 @@ spec:
...
                 name: hocs-queue-config
+            - configMapRef:
+                name: complaint-identity
           env:
             - name: JAVA_OPTS
               value: >
                 -Xms768m -Xmx4096m -XX:+UseG1GC -Djavax.net.ssl.trustStore=/etc/keystore/truststore.jks
                 -Dhttps.proxyHost=hocs-outbound-proxy.{{.KUBE_NAMESPACE}}.svc.cluster.local
-                -Dhttps.proxyPort=31290 
+                -Dhttps.proxyPort=31290
```
Comment (zuzak) [inline kd/deployment.yaml:112]: (this should be a separate commit)

File: kd/identityConfigMap.yaml (hunk 1 above)
Comment (zuzak) [reply kd/identityConfigMap.yaml:8]: Go templating docs are here: https://golang.org/pkg/text/template/
We also use sprig: https://masterminds.github.io/sprig/

**Review by zuzak — APPROVED** [review]
approving; i'd prefer it my way but there's nothing wrong with the code itself.

Remember to squash
